lib/SecretsCrypto/sf-crypto-cipher-session.h

```diff
@@ -0,0 +1,71 @@
+#ifndef SF_CRYPTO_CIPHER_SESSION_H
+#define SF_CRYPTO_CIPHER_SESSION_H
+
+#include <glib-object.h>
+#include <gio/gio.h>
+#include "sf-crypto-manager.h"
+#include "sf-crypto.h"
+#include "sf-crypto-key.h"
+
+#define SF_TYPE_CRYPTO_CIPHER_SESSION (sf_crypto_cipher_session_get_type())
+#define SF_CRYPTO_CIPHER_SESSION(o) (G_TYPE_CHECK_INSTANCE_CAST((o), SF_TYPE_CRYPTO_CIPHER_SESSION, SfCryptoCipherSession))
+#define SF_IS_CRYPTO_CIPHER_SESSION(o) (G_TYPE_CHECK_INSTANCE_TYPE((o), SF_TYPE_CRYPTO_CIPHER_SESSION))
+
+typedef struct SfCryptoCipherSession_ SfCryptoCipherSession;
+typedef struct SfCryptoCipherSessionClass_ SfCryptoCipherSessionClass;
+
+struct SfCryptoCipherSession_ {
+    GObject parent;
+};
+
+struct SfCryptoCipherSessionClass_ {
+    GObjectClass parent_class;
+};
+
+GType sf_crypto_cipher_session_get_type(void);
+
+void sf_crypto_cipher_session_new(SfCryptoManager *manager,
+        GBytes *iv,
+        SfCryptoKey *key,
+        SfCryptoOperation operation,
+        SfCryptoBlockMode block_mode,
+        SfCryptoEncryptionPadding padding,
+        SfCryptoSignaturePadding signature_padding,
+        SfCryptoDigest digest,
+        GHashTable *custom_parameters,
+        const gchar *provider_name,
+        GCancellable *cancellable,
+        GAsyncReadyCallback callback,
+        gpointer user_data);
+SfCryptoCipherSession *sf_crypto_cipher_session_new_finish(GAsyncResult *res, GError **error);
```

Comment:
Are there synchronous versions of those too? In some cases (e.g. a command line app) synchronous calls are significantly more convenient.

Reply:
No synchronous versions are provided, wrappers could be provided with reasonable effort (some similarily as in the tests)